matchtree.go

```diff
@@ -292,12 +293,19 @@ func (t *symbolSubstrMatchTree) prepare(doc uint32) {
 			continue
 		}
 
-		if end <= sections[secIdx].End {
-			t.current[0].symbol = true
-			t.current[0].symbolIdx = uint32(secIdx)
-			trimmed = append(trimmed, t.current[0])
+		if end > sections[secIdx].End {
+			t.current = t.current[1:]
+			continue
+		}
+
+		if t.anchored && !(start == sections[secIdx].Start && end == sections[secIdx].End) {
+			t.current = t.current[1:]
+			continue
 		}
```

Comment:
If the pattern was anchored, check that the match is not just contained in a symbol's range, but matches it exactly.

Reply:
based on this code should the variable name be more something along the lines of `exact`? This is on substr not a regexp, I think anchored can be misinterpreted a bit. There are others parts in the zoekt API where we use the word exact to mean the string must match exactly.